holoviews/core/data/grid.py

```diff
@@ -600,14 +615,20 @@ def range(cls, dataset, dimension):
             expanded = cls.irregular(dataset, dimension)
             column = cls.coords(dataset, dimension, expanded=expanded, edges=True)
         else:
-            column = dataset.dimension_values(dimension)
+            column = cls.values(dataset, dimension, flat=False)
         if column.dtype.kind == 'M':
-            return column.min(), column.max()
+            dmin, dmax = column.min(), column.max()
+            if da and isinstance(column, da.Array):
+                return dmin.compute(), dmax.compute()
+            return dmin, dmax
         elif len(column) == 0:
             return np.NaN, np.NaN
         else:
             try:
-                return (np.nanmin(column), np.nanmax(column))
+                dmin, dmax = (np.nanmin(column), np.nanmax(column))
```

Comment:
This should be lazy as long as the user has NumPy 1.13.0+ and Dask 0.15.0+. If we use the associated Dask Array functions, we can relax the version constraints needed to make this lazy.

Reply:
Okay, I'll use those, although I think we rely on dask>=1.7.0 anyway due to the np.squeeze ``axis`` argument, which is used in a few places.

Reply:
Just to clarify, the reason relying on NumPy ufuncs requires both NumPy and Dask version constraints is because we need the `__array_ufunc__` attribute that NumPy 1.13 added and the Dask 0.15.0 version that defines `__array_ufunc__ ` for Dask Arrays. If we use the Dask ufuncs, this doesn't occur.